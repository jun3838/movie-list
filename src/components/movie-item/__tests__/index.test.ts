import { mount } from '@vue/test-utils'
import MovieItem from '../index.vue'

describe('should render movie item correctly', () => {
  test('without favourite', () => {
    const wrapper = mount(MovieItem, {
      props: {
        title: 'Avatar',
        year: 2000,
        imdbId: 'abc123',
        favourites: []
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('with favourite', () => {
    const wrapper = mount(MovieItem, {
      props: {
        title: 'Avatar',
        year: 2000,
        imdbId: 'abc123',
        favourites: [{
          title: 'Avatar',
          year: 2000,
          imdbId: 'abc123'
        }]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('methods', () => {
  describe('getIconSymbol', () => {
    test('should return #heart', () => {
      const wrapper = mount(MovieItem, {
        props: {
          title: 'Avatar',
          year: 2000,
          imdbId: 'abc123',
          favourites: []
        }
      })

      const result = wrapper.vm.getIconSymbol('abc123')
      expect(result).toEqual('#heart')
    })

    test('should return #heart-fill', () => {
      const wrapper = mount(MovieItem, {
        props: {
          title: 'Avatar',
          year: 2000,
          imdbId: 'abc123',
          favourites: [{
            title: 'Avatar',
            year: 2000,
            imdbId: 'abc123'
          }]
        }
      })

      const result = wrapper.vm.getIconSymbol('abc123')
      expect(result).toEqual('#heart-fill')
    })
  })
})
