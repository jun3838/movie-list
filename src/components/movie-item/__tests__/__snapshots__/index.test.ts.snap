// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render movie item correctly with favourite 1`] = `
"<div class=\\"thumbnail relative rounded-md flex flex-col p-3 rounded-md hover:bg-gray-100 cursor-pointer\\"><span class=\\"title text-lg font-medium\\">Avatar</span><span class=\\"year text-sm text-zinc-500\\">Year: 2000</span><span class=\\"imdb-id text-xs text-gray-400\\">imdb ID: abc123</span><svg class=\\"icon w-6 h-6 absolute right-2 bottom-2\\">
    <use xlink:href=\\"#heart-fill\\"></use>
  </svg></div>"
`;

exports[`should render movie item correctly without favourite 1`] = `
"<div class=\\"thumbnail relative rounded-md flex flex-col p-3 rounded-md hover:bg-gray-100 cursor-pointer\\"><span class=\\"title text-lg font-medium\\">Avatar</span><span class=\\"year text-sm text-zinc-500\\">Year: 2000</span><span class=\\"imdb-id text-xs text-gray-400\\">imdb ID: abc123</span><svg class=\\"icon w-6 h-6 absolute right-2 bottom-2\\">
    <use xlink:href=\\"#heart\\"></use>
  </svg></div>"
`;
